tests/frontier/blockchain_st_test_helpers.py

```diff
@@ -90,7 +91,19 @@ def load_test(test_dir: str, test_file: str, network: str) -> Dict[str, Any]:
     test_name = os.path.splitext(pure_test_file)[0]
     path = os.path.join(test_dir, test_file)
     with open(path, "r") as fp:
-        json_data = json.load(fp)[f"{test_name}_{network}"]
+        data = json.load(fp)
+
+        # Some newer test files have for example _d0g0v0_
+        # between test_name and network
+        keys_to_search = re.compile(
+            f"{test_name}_d[0-9]g[0-9]v[0-9]_{network}"
```

Comment:
Would it be a good idea to [`re.escape`](https://docs.python.org/3/library/re.html#re.escape) these?

Reply:
All the regex meta-characters that we have passed in will be appropriately handled by `re.compile`. And no meta-characters appear in the `test_name` as far as I have seen. So, we should be ok.

Reply:
Instead of using the same function for legacy tests and the new tests, would it make sense to have a separate function for handling the new tests?